pkg/testlib/testlib.go

```diff
@@ -32,26 +31,39 @@ import (
 )
 
 const (
-	defaultModel    = "testdata/default"
+	DefaultModel    = "testlib/testdata/default"
 	defaultDC       = "DC0"
 	defaultVMPath   = "/DC0/vm/"
 	defaultHost     = "H0"
-	defaultHostId   = "host-24" // Generated by vcsim
+	DefaultHostId   = "host-24" // Generated by vcsim
 	defaultHostPath = "/DC0/host/DC0_"
 )
 
-type simulatedVM struct {
-	name, uuid string
+var (
+	NodeProperties = []string{"config.extraConfig", "config.flags", "config.version", "runtime.host"}
+)
+
+type SimulatedVM struct {
+	Name, UUID string
 }
 
 var (
 	// Virtual machines generated by vSphere simulator. UUIDs look generated, but they're stable.
-	defaultVMs = []simulatedVM{
+	defaultVMs = []SimulatedVM{
 		{"DC0_H0_VM0", "265104de-1472-547c-b873-6dc7883fb6cb"},
 		{"DC0_H0_VM1", "12f8928d-f144-5c57-89db-dd2d0902c9fa"},
 	}
 )
 
+func init() {
+	_, filename, _, _ := goruntime.Caller(0)
+	dir := path.Join(path.Dir(filename), "..")
+	err := os.Chdir(dir)
```

Comment:
So I feel less strongly about this.  We have done same here - https://github.com/openshift/vmware-vsphere-csi-driver-operator/blob/master/pkg/operator/testlib/vcsim_framework.go#L50 and nobody has even noticed. 

Having the testlib file `Chdir`  means, we can assume that all tests are executing with `PWD` (or `CWD` if you will) of `pkg` as root and it ensures that no matter how deeply nested a test is, it can always access model files uniformly. 
 

Overriding `DefaultModel` here will mean that, depending on where the actual test is - it must calculate relative path to model files or else all bets are off. IMO - it much cleaner to have test execute from same `PWD`.

Reply:
The big issue is not just finding of model directory, but the way path of folders etc is hardcoded in generated testdata -  https://github.com/openshift/vsphere-problem-detector/blob/master/pkg/check/testdata/default/0080-Datastore-%252Ftmp%252Fgovcsim-DC0-LocalDS_2-363093114%2540folder-5.xml#L80

So yeah specifying different relative path to model directory from tests will work, but the problem is - relative path hardcoded inside testdata will not match, model directory and hence no resource will be returned for certain simulated API calls. 

For example, here is what I did for folder_test.go:

```

func TestCheckFolderPermissions(t *testing.T) {
	// Very simple test, no error cases

	// Stage
	kubeClient := &testlib.FakeKubeClient{
		Infrastructure: testlib.Infrastructure(),
	}

	modelPath := "../testlib/testdata/default"
	ctx, cleanup, err := SetupSimulator(kubeClient, modelPath)
}
```

This causes folder_test.go to fail, while all other tests pass.